### PR TITLE
feat(storage_profile): add support for CSI arguments

### PR DESCRIPTION
**Pull request by aescrob**
## Describe your changes
Since settings are supported by terraform-azurerm-provider from v3.35.0 on, allowing to disable CSI driver one by one.

## Issue number

#000

## Checklist before requesting a review
- [x] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [x] I have executed pre-commit on my machine
- [x] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

**Comment by aescrob** [comment]
> @microsoft-github-policy-service agree [company="{swisspost}"]

File: variables.tf
Comment (lonegunmanb) [inline variables.tf:240]: I think we've got a typo in this description.

File: main.tf
Comment (lonegunmanb) [inline main.tf:222]: Could we make this block a `dynamic` block and add a new toggle variable so we can turn it on and off like we've done for [`aci_connector_linux`](https://github.com/Azure/terraform-azurerm-aks/blob/master/main.tf#L90-L96)?

File: variables.tf
Comment (lonegunmanb) [inline variables.tf:536]: According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L1064-L1068), the `default` should be `false`.

File: variables.tf
Comment (lonegunmanb) [inline variables.tf:542]: According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L1069-L1073), the `default` should be `true`.

File: variables.tf
Comment (lonegunmanb) [inline variables.tf:548]: According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L1074-L1082), the `default` should be `"v1"`.

File: variables.tf
Comment (lonegunmanb) [inline variables.tf:554]: According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L1083-L1087), the `default` should be `true`.

File: variables.tf
Comment (lonegunmanb) [inline variables.tf:560]: According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L1088-L1092), the `default` should be `true`.

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @aescrob for opening this pr! We have a CI pipeline in this repo, would you please follow the [instruction](https://github.com/Azure/terraform-azurerm-aks#pre-commit--pr-check--test) in the readme, run `pre-commit` job on you machine and try again? Thanks.

File: variables.tf
Comment (aescrob) [reply variables.tf]: Fixed

File: variables.tf
Comment (aescrob) [reply variables.tf]: Changed

File: variables.tf
Comment (aescrob) [reply variables.tf]: Changed

File: variables.tf
Comment (aescrob) [reply variables.tf]: Changed

File: variables.tf
Comment (aescrob) [reply variables.tf]: Changed

File: variables.tf
Comment (aescrob) [reply variables.tf]: Changed

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Apology for another change request @aescrob, I wasn't able to figure them out at the first review.

Since we use terraform-docs to generate document from variable and output descriptions, would you please help us by back-quoting values inside the description? Thanks!

And one more thing, our pipeline was broken because of Checkov has added some new policies, so please wait until #281 was merged then rebase to the latest master branch, thanks for your understanding and cooperation!

File: variables.tf
Comment (lonegunmanb) [inline variables.tf:554]: Could we add `nullable = false` here?

**Review by lonegunmanb — CHANGES_REQUESTED** [review]

**Comment by lonegunmanb** [comment]
Hi @aescrob, since the master branch now should been fixed, would you please rebase to the latest master branch and try again? Thanks.

**Comment by aescrob** [comment]
> @aescrob please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="swisspost"]

**Comment by aescrob** [comment]
> @aescrob please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="swisspost"

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Hi @aescrob, this pr has passed the e2e test and almost LGTM, only one extra commit with declaring `var.storage_profile_enabled`'s `nullable = false` would make this pr been merged, thanks for your understanding.

**Comment by lonegunmanb** [comment]
Hello @aescrob, I'm sorry we've met an issue on our pipeline so we have to upgrade the Github action yaml files in this repo, would you please merge your branch with the latest master branch so we can run the e2e test? Apology for the inconvenience, we're still refactoring the pipeline.

**Comment by aescrob** [comment]
Oh...i am so sad to not made it into v6.3.0.
Anyway, thank you @lonegunmanb for your feedbacks

**Comment by lonegunmanb** [comment]
> Oh...i am so sad to not made it into v6.3.0. Anyway, thank you @lonegunmanb for your feedbacks

No worries @aescrob , I'll release a new version for you once this pr has been merged.

**Review by lonegunmanb — APPROVED** [review]
Thanks @aescrob, LGTM! 🚀

**Pull request merged**
